apps/web/components/settings/CalendarSwitch.tsx

```diff
@@ -68,8 +70,9 @@ export function CalendarSwitch(props: {
         key={props.externalId}
         name="enabled"
         label={props.title}
-        defaultChecked={props.isSelected}
+        checked={isSelected}
```

Comment:
We should also update the CalendarSwitch component here https://github.com/calcom/cal.com/blob/2602a66f7c2a2d252c4b0a6d1cad50009e4705ce/apps/web/components/apps/CalendarListContainer.tsx#L83

Also, it would be better to maintain a single component as they do the same work.

Reply:
But it's still has some delay for like a 1-2 s delay? Is this a fine behaviour? I mean we can use a useState and useEffect (invalidating in connectedCalendars in useEffect and if requests fails we can disable it) for immediate affects.

Reply:
It should be faster in production. Also, we can implement optimistic UI if we feel it's quite slow.